### ISSUE TITLE
Pre-commit CI - GitHub action

### ISSUE THREAD

**Issue by rathorevaibhav** [ColoredCow-Portal/portal#204]
**Depedency**
Requires #203 to be finished first

**Describe the task**
Once pre-commit (or some other linting library is in place), we need to automate it with GitHub Actions.

**Expected behavior**
1. Pre-commit checks should run whenever a new pull request is opened against master branch
2. Pre-commit checks should run whenever a 
3. Pre-commit status badge added to readme

**References**
1. Check pre-commit GitHub action in Plio: https://github.com/avantifellows/plio-frontend/blob/master/.github/workflows/ci.yml
2. Check Plio readme for pre-commit badge
